cephfs/file_test.go

```diff
@@ -615,3 +615,95 @@ func TestFallocate(t *testing.T) {
 		assert.Error(t, err)
 	})
 }
+
+func TestFlock(t *testing.T) {
+	mount := fsConnect(t)
+	defer fsDisconnect(t, mount)
+
+	t.Run("validate", func(t *testing.T) {
+		f := &File{}
+		err := f.Flock(LockSH, 1010)
+		assert.Error(t, err)
+	})
+
+	t.Run("validateOperation", func(t *testing.T) {
+		fname := "file.txt"
+		f, err := mount.Open(fname, os.O_RDWR|os.O_CREATE, 0666)
+		assert.NoError(t, err)
+		assert.NotNil(t, f)
+		defer func() {
+			assert.NoError(t, f.Close())
+			assert.NoError(t, mount.Unlink(fname))
+		}()
+		err = f.Flock(1037, 1010)
```

Comment:
instead of the fictional value `1037`, you can pass `LockEX|LockSH|LockUN` or some invalid combination like that